tractor/_forkserver_hackzorz.py

```diff
@@ -0,0 +1,284 @@
+"""
+This is near-copy of the 3.8 stdlib's ``multiprocessing.forkserver.py``
+with some hackery to prevent any more then a single forkserver and
+semaphore tracker per ``MainProcess``.
+"""
+import os
+import socket
+import signal
+import struct
+import sys
+import errno
+import selectors
+import warnings
+
+from multiprocessing import (
+    forkserver, semaphore_tracker, spawn, process, util
+)
+from multiprocessing.forkserver import (
+        ForkServer, MAXFDS_TO_SEND, write_unsigned,
+        # _serve_one,
+)
+from multiprocessing.context import reduction
+
+
+# taken from 3.8
+SIGNED_STRUCT = struct.Struct('q')     # large enough for pid_t
+
+
+class AdultForkServer(ForkServer):
+
+    def connect_to_new_process(self, fds):
+        '''Request forkserver to create a child process.
+
+        Returns a pair of fds (status_r, data_w).  The calling process can read
+        the child process's pid and (eventually) its returncode from status_r.
+        The calling process should write to data_w the pickled preparation and
+        process data.
+        '''
+        # self.ensure_running()  # treat our users like adults expecting
+        # them to spawn the server on their own
+        if len(fds) + 4 >= MAXFDS_TO_SEND:
```

Comment:
As an aside, what's the fourth descriptor?

Reply:
No idea - ain't my code. There's 2 pipes and a socket iirc.

Reply:
likely 3 pipes: stdin, stdout, stderr.